### PR TITLE
Update logic.md

### PR DESCRIPTION
**Pull request by jba**
I really like the informal tone and the clarity of this doc.

I rephrased some things, fixed some typos, and added comments where I thought things could be clearer.

### CODE REVIEW

File: content/en/docs/concepts/logic.md [hunk 1]
```diff
@@ -276,11 +275,14 @@ graph TD
 
 There are actually values between top and the basic types.
 The `|` operator in CUE allows one to define "sum types" like `int | string`.
-The same operator can also be used to describe what is called "enums"
+The same operator can also be used to describe what are called "enums"
 in other languages, for instance, `1 | 2 | 3`.
 To CUE these are the same thing.
+<!-- (jba) The rest of this paragraph is obscure. What are "the two" things being mixed? How
+does the default example show that? -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:282]: Two things: the two levels of abstractions: types and values.

File: content/en/docs/concepts/logic.md [hunk 2]
```diff
@@ -276,11 +275,14 @@ graph TD
...
 You can also mix the two, as in `*1 | int`to define defaults (marked by `*`),
 or mix it with expressions like `*pet.species | "cat"`.
+<!-- (jba) The reader doesn't have enough info to understand this disjunction, so they can't map it to null coalescing even if they knew what that was. -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:285]: would `*pet["species"]` be any better?

File: content/en/docs/concepts/logic.md [hunk 3]
```diff
@@ -356,7 +358,9 @@ in the older version.
 
 With optional fields it gets a bit more subtle, but basically,
 an instance may change an optional field to required, but not remove it.
-The backwards compatibility metaphore applies here as well.
+The backwards compatibility metaphor applies here as well. 
+<!-- (jba) It does apply, but in reverse order. When you write "an instance may change..." I immediately thought of the instance
+as the new API (since it is altering something that exists), but that is backwards. -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:363]: That is correct.

**Review by mpvl — COMMENTED** [review]
partially reviewed.

File: content/en/docs/concepts/logic.md [hunk 4]
```diff
@@ -400,6 +404,11 @@ An important thing to note is that, unlike for required fields,
 conflicting values for an optional fields do not cause a struct to be faulty.
 This definition was a result from fitting the notion of closed structs into
 the value lattice.
+<!-- (jba) First mention of closed; no one knows what it means. 
+Rest of paragraph is a bit obscure; explain in plain English instead.
+E.g. "If an optional field's value is innnconsistent, we can just drop the field from the struct.
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md]: Ah, you use a MacBook as well I see.

File: content/en/docs/concepts/logic.md [hunk 5]
```diff
@@ -475,14 +487,25 @@ But because types are values in CUE, TypeScript's concepts of
 `null`, bottom (`_|_`), and optional fields,
 resulting in a somewhat simpler model.
 
+<!-- (jba) Without more discussion and examples, the whole null section is confusing. -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:490]: Okay, I'll think about.

File: content/en/docs/concepts/logic.md [hunk 6]
```diff
@@ -492,30 +515,36 @@ it is guaranteed that this will be the final result.
...
 for a concrete value is not over.
 In other words, an instance may never violate the constraints of its parent.
-This property makes it very hard to inadvertendly make false conclusions in CUE.
+This property makes it very hard to inadvertently make false conclusions in CUE.
 Default values do not change this property; they syntactically appear as
 non-concrete values.
 CUE also bails out and requires explicit values if two conflicting defaults
 are specified for the same field, again limiting the search space.
+<!-- (jba) There's no actual searching going on in the implementation (IIUC), so using the term "search
+space" is confusing. -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:524]: The search applies to users trying to track dow the origin of a value, not CUE's evaluation.

File: content/en/docs/concepts/logic.md [hunk 7]
```diff
@@ -492,30 +515,36 @@ it is guaranteed that this will be the final result.
...
-it can be time consuming and,
-especially in a pressured situation,
+it can be time-consuming and,
+especially when under pressure,
 very tempting to skip following complicated inheritance chains,
-double check a configuration file specifying overlay order,
+double-check a configuration file specifying overlay order,
 or look for a file that is lexically sorted after the one under consideration.
+<!-- (jba) I think it's worth making the more general point that order-independence
+makes life much simpler. Compare with functional vs. imperative languages? -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:539]: Hadn't thought of that analogy before. Will find some wording for it.

File: content/en/docs/concepts/logic.md [hunk 8]
```diff
@@ -492,30 +515,36 @@ it is guaranteed that this will be the final result.
...
 
 So there is a clear benefit to having fully expanded configurations
 over such override methods.
 CUE simulates that benefit by guaranteeing that any observed field value
 holds for the final result.
 If a user makes a false assumption that a concrete (default)
 value is the last in a chain,
+<!-- (jba) "last in a chain" makes it sound like order matters -->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:547]: `If the user makes the false assumption that no concrete value is specified to discard the default value,`

perhaps?

File: content/en/docs/concepts/logic.md [hunk 9]
```diff
@@ -642,6 +671,11 @@ that do not depend on each other.
 
 The values lattice brings CUE another advantage: the ability to reason about
 values, schema, and constraints.
+<!-- (jba) I think "schema" is confusing as the plural. Choose either "schemas"
+or "schemata". I think the former is, sadly, preferable to modern ears.
+(This applies to several instances of "schema" above too.) 
+-->
```
Comment (mpvl) [inline content/en/docs/concepts/logic.md:677]: What's next, a bottoms-up model? I prefer schemata, but you are probably right about shemas being better.

**Comment by jba** [comment]
I'm confused. I just did `git clone https://github.com/cuelang/cuelang.org` and I'm seeing these changes on master. Were they somehow made independently of this PR?
